rfcs/0019-maintainers.md

```diff
@@ -0,0 +1,95 @@
+---
+feature: maintainers-file
+start-date: 2017-10-28
+author: Maarten Hoogendoorn (@moretea)
+co-authors: (find a buddy later to help our with the RFC)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Currently, nixpkgs does not have explicit maintainers for anything besides
+the packages themselves. Introducing a maintainers file enables us to mark
+maintainers for the remaining parts.
+
+# Motivation
+[motivation]: #motivation
+
+<!--  Why are we doing this? -->
+There is no explicit (machine-readable) description of who maintains what.
+
+Currently, a bot is used to ping people based on the heuristic of who authored
+a git commit that that have touched a file at least once.  
+This does not accurately track the current maintainers.
+For example, if someone took over maintenance of the NixOS module system, nbp 
+will still get pinged all the time, because he has created most of the files
+in there.
+
+Furthermore, there is only the rather black-and-white distinction between
+people who have commit access, and those that do not. A maintainers file
+might open further automation to delegate merge access.
+
+<!-- What use cases does it support? -->
+With a maintainers file, we can ping the correct maintainer(s) on a PR.
```

Comment:
We can already. By evaluating Nixpkgs we see which derivations are affected and can thus ping those maintainers. We just need  a bot for it.